pkg/controller.v1alpha3/suggestion/suggestion_controller_util.go

```diff
@@ -0,0 +1,36 @@
+package suggestion
+
+import (
+	"context"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r *ReconcileSuggestion) createOrUpdateDeployment(deploy *appsv1.Deployment) (*appsv1.Deployment, error) {
+	foundDeploy := &appsv1.Deployment{}
+	err := r.Get(context.TODO(), types.NamespacedName{Name: deploy.Name, Namespace: deploy.Namespace}, foundDeploy)
+	if err != nil && errors.IsNotFound(err) {
+		log.Info("Creating Deployment", "namespace", deploy.Namespace, "name", deploy.Name)
+		err = r.Create(context.TODO(), deploy)
+		return nil, err
+	} else if err != nil {
+		return nil, err
+	}
+	return foundDeploy, nil
+}
+
+func (r *ReconcileSuggestion) createOrUpdateService(service *corev1.Service) (*corev1.Service, error) {
```

Comment:
Aren't we updating the service here?

Reply:
No idea about if we have such a use case now.

Reply:
Ok.  we can rename to ReconcileService?

Reply:
Sure